src/scumm-8.p8

```diff
@@ -2605,84 +2317,65 @@ function recalc_bounds(obj, w, h, cam_off_x, cam_off_y)
- insert(frontier, start, 0)
- came_from[vectoindex(start)] = nil
- cost_so_far[vectoindex(start)] = 0
-
- while #frontier > 0 and #frontier < 1000 do
- 	-- pop the last element off a table
-	local top = frontier[#frontier]
-	del(frontier,frontier[#frontier])
-	current = top[1]
-
-  if vectoindex(current) == vectoindex(goal) then
-   break
-  end
+	local frontier, came_from, cost_so_far, lowest_dist, lowest_dist_node, current = {}, {}, {}
+	insert(frontier, start, 0)
+	-- came_from[vectoindex(start)] = nil
+	cost_so_far[vectoindex(start)] = 0
 
-  --local neighbours = getneighbours(current)
-	local neighbours = {}
-	for x = -1, 1 do
-		for y = -1, 1 do
-			if x == 0 and y == 0 then 
-				--continue 
-			else
-				local chk_x = current[1] + x
-				local chk_y = current[2] + y
+	while #frontier > 0 and #frontier < 1000 do
+		-- pop the last element off a table
+		current = frontier[#frontier][1]
+		del(frontier, frontier[#frontier])
+
+		if (vectoindex(current) == vectoindex(goal)) break
+
+		--local neighbours = getneighbours(current)
+		local neighbours = {}
+		for x = -1, 1 do
+			for y = -1, 1, x == 0 and 2 or 1 do
+				local chk_x, chk_y = current[1] + x, current[2] + y
 
-				-- diagonals cost more
-				if abs(x) != abs(y) then cost=1 else cost=1.4 end
-				
 				if chk_x >= room_curr.map[1] and chk_x <= room_curr.map[1] + room_curr.map_w
-				 and chk_y >= room_curr.map[2] and chk_y <= room_curr.map[2] + room_curr.map_h
-				 and is_cell_walkable(chk_x, chk_y)
+					and chk_y >= room_curr.map[2] and chk_y <= room_curr.map[2] + room_curr.map_h
+					and is_cell_walkable(chk_x, chk_y)
 				-- squeeze check for corners
-				 and ((abs(x) != abs(y)) 
-						or is_cell_walkable(chk_x, current[2]) 
+					and ((abs(x) != abs(y))
+						or is_cell_walkable(chk_x, current[2])
 						or is_cell_walkable(chk_x - x, chk_y)
-						or enable_diag_squeeze) 
-				then
-					-- add as valid neighbour
-					add( neighbours, {chk_x, chk_y, cost} )
-				end
-			end
-		end
-	end
-	-- --------------
-
-  for next in all(neighbours) do
-   local nextindex = vectoindex(next)
-   local new_cost = cost_so_far[vectoindex(current)] + next[3] -- add extra costs here
-
-   if not cost_so_far[nextindex]
-	  or new_cost < cost_so_far[nextindex]
+						or enable_diag_squeeze)
 	 then
+					-- process a valid neighbor
+					local next = {chk_x, chk_y}
+					local nextindex = vectoindex(next)
+					local new_cost = cost_so_far[vectoindex(current)] + (x * y == 0 and 1 or 1.414) -- diagonals cost more
+
+					if not cost_so_far[nextindex] or new_cost < cost_so_far[nextindex] then
 	 	cost_so_far[nextindex] = new_cost
-		-- diagonal movement - assumes diag dist is 1, same as cardinals
-		local h = max(abs(goal[1] - next[1]), abs(goal[2] - next[2]))
-		local priority = new_cost + h
-    insert(frontier, next, priority)
+						local h = max(abs(goal[1] - chk_x), abs(goal[2] - chk_y)) + min(abs(goal[1] - chk_x), abs(goal[2] - chk_y)) * .414
```

Comment:
I think this is a more accurate heuristic cost, assumes walking diagonally to get axis aligned with the target then walking straight to it.